cms/djangoapps/contentstore/views/item.py

```diff
@@ -145,12 +146,25 @@ def xblock_handler(request, usage_key_string):
             accept_header = request.META.get('HTTP_ACCEPT', 'application/json')
 
             if 'application/json' in accept_header:
+                store = modulestore()
                 fields = request.GET.get('fields', '').split(',')
+                format = request.GET.get('format', '')
                 if 'graderType' in fields:
                     # right now can't combine output of this w/ output of _get_module_info, but worthy goal
                     return JsonResponse(CourseGradingModel.get_section_grader_type(usage_key))
+                elif format == 'concise':
+                    xblock = _get_xblock(usage_key, request.user)
+                    course = store.get_course(xblock.location.course_key)   # pylint: disable=no-member
+                    course_outline = create_xblock_info(
+                        course,
+                        include_child_info=True,
+                        include_children_predicate=lambda xblock: xblock.has_children,
+                        is_concise=format
+                    )
+                    ancestor_info = _create_xblock_ancestor_info(xblock, course_outline, is_concise=format)
```

Comment:
My recommendation would be to use the ``course_handler``, as that is the natural API for returning the course outline. I don't recommend extending it to support additionally returning ancestor info for a block, as IMO that isn't a natural behavior for the API.

One option would be to pass the current parent down to the client in Mako, because its location is known. Then a new request doesn't need to be issued when moving it. I suppose this leaves a window of error where someone else could move the item but that seems unlikely.

Another option would be to issue two requests: one to get the course outline, and a second one to get the current location. In some ways this is better because the two pieces of information are used by different parts of the UI, so it improves the modularity.